src/elfpy/simulators.py

```diff
@@ -213,14 +214,17 @@ def run_simulation(self) -> None:
         self.start_time = time_utils.current_datetime()
         for day in range(0, self.config.simulator.num_trading_days):
             self.day = day
+            self.market.market_state.vault_apr = self.random_variables.vault_apr[self.day]
             # Vault return can vary per day, which sets the current price per share
             if self.day > 0:  # Update only after first day (first day set to init_share_price)
-                self.market.market_state.share_price += (
-                    self.random_variables.vault_apr[self.day]  # current day's apy
-                    / 365  # convert annual yield to daily
-                    * self.market.market_state.init_share_price  # APR, apply return to starting price (no compounding)
-                    # * self.market.share_price # APY, apply return to latest price (full compounding)
+                delta = MarketDeltas(
+                    d_share_price=(
+                        self.market.market_state.vault_apr  # current day's apy
+                        / 365  # convert annual yield to daily
+                        * self.market.market_state.init_share_price  # apply return to starting price (no compounding)
```

Comment:
I think it probably makes sense to use compounding (or at least make this configurable). 

See the "Calculating Accrued Interest" section of this document: https://docs.compound.finance/v2/. Since Compound has so much activity, its users experience a close approximation of compounding every block. I spent some time reading through the Compound III contracts, and I believe this to be the case for those contracts too.

Reply:
This was the previously-existing formula; I didn't modify it in this PR. I guess I should add a flag to config.simulator? Or do you think there is no scenario where one might want a "no compounding" option?

Reply:
This is a case in which I think having both options is good, so a flag would be good. One should use `self.market.market_state.share_price` and the other should use `init_share_price` (as is currently the case).